enterprise/server/byte_stream_server_proxy/byte_stream_server_proxy.go

```diff
@@ -291,10 +300,22 @@ func (s *ByteStreamServerProxy) Write(stream bspb.ByteStream_WriteServer) error
 	stream = meteredStream
 	var err error
 	if authutil.EncryptionEnabled(ctx, s.authenticator) && !s.supportsEncryption(ctx) {
+		if config.UnconditionallyCompressWrites() {
+			stream, err = s.toCompressingStream(stream)
+			if err != nil {
+				return err
+			}
+		}
 		err = s.writeRemoteOnly(ctx, stream)
 	} else if proxy_util.SkipRemote(ctx) {
 		err = s.writeLocalOnly(stream)
```

Comment:
is this to avoid compressing then decompressing to verify the digest? if so, i think the dualWrite path will be affected by this too, so could you add a TODO to fix that?